subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/GradleLogEventListener.kt

```diff
@@ -0,0 +1,48 @@
+package com.avito.android.build_verdict.internal
+
+import org.gradle.api.logging.LogLevel
+import org.gradle.api.logging.StandardOutputListener
+import org.gradle.internal.logging.events.CategorisedOutputEvent
+import org.gradle.internal.logging.events.LogEvent
+import org.gradle.internal.logging.events.OutputEvent
+import org.gradle.internal.logging.events.OutputEventListener
+import org.gradle.internal.logging.events.StyledTextOutputEvent
+import org.gradle.internal.operations.OperationIdentifier
+
+internal class GradleLogEventListener(
+    private val acceptedLevel: LogLevel = LogLevel.ERROR,
+    /**
+     * "org.gradle.api.Task" - org.gradle.api.internal.AbstractTask.BUILD_LOGGER
+     * "system.err" - org.gradle.internal.logging.source.DefaultStdErrLoggingSystem
+     */
+    private val acceptedCategories: List<String> = listOf("org.gradle.api.Task", "system.err"),
+    private val listeners: Map<OperationIdentifier, StandardOutputListener>
+) : OutputEventListener {
+
+    override fun onOutput(event: OutputEvent) {
+        when (event) {
+            is CategorisedOutputEvent -> {
+                if (event.logLevel == acceptedLevel && event.category in acceptedCategories) {
+                    when (event) {
```

Comment:
Deep nesting. How about extracting to a function?

Reply:
Done